src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeController.java

```diff
@@ -41,4 +54,26 @@ public EnvelopeMetadataResponse getProcessedEnvelopesByJurisdiction(
             envelopeRetrieverService.getProcessedEnvelopesByJurisdiction(serviceName)
         );
     }
+
+    @PutMapping(path = "/{id}/status")
+    @ApiOperation("Updates the status of given envelope")
+    @ApiResponse(code = 204, message = "Envelope updated")
+    public ResponseEntity<Void> updateStatus(
+        @PathVariable UUID id,
+        @RequestBody StatusUpdate statusUpdate,
+        @RequestHeader(name = "ServiceAuthorization", required = false) String serviceAuthHeader
+    ) {
+        String serviceName = authService.authenticate(serviceAuthHeader);
+
+        // TODO: move to service
+        switch (statusUpdate.status) {
+            case CONSUMED:
+                envelopeUpdateService.markAsConsumed(id, serviceName);
+                break;
+            default:
+                throw new IllegalStateException("No code for handling status: " + statusUpdate.status);
```

Comment:
Is IllegalStateException the right kind of exception to be used here ? Could we use InvalidStatusException or something similar ?

Reply:
this branch is never reached due to single enum being present in enum class